ci/scripts/python_wheel_manylinux_build.sh

```diff
@@ -71,6 +71,10 @@ echo "=== (${PYTHON_VERSION}) Building Arrow C++ libraries ==="
 : ${VCPKG_FEATURE_FLAGS:=-manifests}
 : ${VCPKG_TARGET_TRIPLET:=${VCPKG_DEFAULT_TRIPLET:-x64-linux-static-${CMAKE_BUILD_TYPE}}}
 
+if [[ "$(uname -m)" == arm* ]] || [[ "$(uname -m)" == aarch* ]]; then
+    export ARROW_EXTRA_CMAKE_FLAGS="-DARROW_JEMALLOC_LG_PAGE=14"
```

Comment:
Can you add a comment explaining why this is necessary?

Reply:
Should it be 12 (4K)?

Reply:
`--with-lg-page=16` fixes issue on 64k page system. https://github.com/apache/arrow/issues/10929#issuecomment-899950190

I tested on 4k page system with the image built from this pr `--with-lg-page=14`, it work okay.

According to https://github.com/jemalloc/jemalloc/issues/467#issuecomment-294383178, looks we can set `--with-lg-page=16` and the binary should work on both 4k and 64k page arm64 system.

Reply:
Updated to `16`, though the conda-forge recipe uses `14`.

Reply:
Arm64 does support 4k/16k/64k page sizes. 16k is rarely used and looks deprecated AFAIK.